is_cartesian isolate X and Y behaviour

AdamLay opened the pull request:
When running on cartesian machines, only home the relevant axes while validating endstop variance and while running the move tests.

Anonoei commented:
Everything looks good. Haven't had a chance to test it yet. I'll try to do it today - won't be able to test with my printers for a couple weeks

Anonoei commented:
@AdamLay 
Only issue is DiagX and DiagY need isolate_xy in their Init method (even though they don't use it). After you add that I can merge. Thanks for this addition!
`def Init(self, axis_limits, margin, isolate_xy):`

AdamLay commented:
Thanks @Anonoei - I've added the params as `_`, not sure if this is the done thing in Python to be honest but I wanted to indicate the params are being discarded (we have no use for DiagX and DiagY on cartesian/CoreXZ - would also probably good to remove these as valid axes)
Also added quick change to amend the default axes for isolate_xy printers as well due to above reason! :) thanks

Anonoei commented:
@AdamLay thanks! I merged this PR. 
_ is the correct name to indicate a variable that isn't used. It's typically used as an empty iterator, like 'for _ in range(0, 10)'

Thanks again for contributing!
